components/molecule/collapsible/src/index.js

```diff
@@ -0,0 +1,127 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+
+const BASE_CLASS = 'sui-MoleculeCollapsible'
+const CONTENT_CLASS = `${BASE_CLASS}-content`
+const CONTAINER_BUTTON_CLASS = `${BASE_CLASS}-container`
+const BUTTON_CLASS = `${BASE_CLASS}-btn`
+const ICON_CLASS = `${BASE_CLASS}-icon`
+const MIN_HEIGHT = 100 // px
+
+class MoleculeCollapsible extends Component {
+  constructor(props) {
+    super(props)
+    this.childrenContainer = React.createRef()
+    this.state = {collapsed: true, showButton: true}
+    this.contentStyles = {
+      height: `${MIN_HEIGHT}`
+    }
+  }
+
+  toggleCollapse = () => {
+    const {collapsed, showButton} = this.state
+    const {onClose, onOpen} = this.props
+    if (showButton) {
+      if (collapsed) {
+        this.setState({collapsed: false})
+        onOpen()
+      } else {
+        this.setState({collapsed: true})
+        onClose()
+      }
+    }
+  }
+
+  componentDidMount() {
+    const offsetHeight = this.childrenContainer.current.offsetHeight
+    this.setState({showButton: offsetHeight >= MIN_HEIGHT})
+  }
+
+  render() {
+    const {collapsed, showButton} = this.state
+    const {children, height, icon, showText, hideText, hasGradient} = this.props
+    const wrapperClassName = cx(`${BASE_CLASS}`, {
+      [`${BASE_CLASS}--gradient`]: hasGradient,
+      [`${BASE_CLASS}--collapsed`]: collapsed
```

Comment:
From my point of view, `collapsed` is much more an state of the component than a modifier, regarding the SUIT CSS conventions: https://github.com/suitcss/suit/blob/master/doc/naming-conventions.md#componentnameis-stateofcomponent, so it should be used as that: `.is-collapsed`. For `gradient` I'm not sure if it could be treated the same way, but in case it's used as modifier, it would be better named `--withGradient`.